### PR TITLE
EQSIM benchmark code and fixes

### PR DESCRIPTION
**Pull request by houjun**
# Related Issues / Pull Requests

List all related issues and/or pull requests if there are any.

# Description
- Add EQSIM-related benchmark code
- Add (partial) support for ZFP compression
- Fix an issue with data querying
- Fix an issue with >1GB reads
- Fix an issue with 2d/3d reads
- Fix an issue with Perlmutter CI module load

# What changes are proposed in this pull request?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected; for instance, examples in this repository must be updated too)
- [x] This change requires a documentation update

# Checklist:

- [ ] My code modifies existing public API, or introduces new public API, and I updated or wrote docstrings
- [x] I have commented my code
- [x] My code requires documentation updates, and I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by houjun** [comment]
@jeanbez @zhangwei217245

File: src/api/pdc_region/pdc_region_transfer.c [hunk 1]
```diff
@@ -1787,25 +1787,29 @@ release_region_buffer(char *buf, uint64_t *obj_dims, int local_ndim, uint64_t *l
     if (local_ndim == 2) {
         if (access_type == PDC_READ) {
             ptr = new_buf;
-            for (i = 0; i < local_size[0]; ++i) {
-                memcpy(buf + ((local_offset[0] + i) * obj_dims[1] + local_offset[1]) * unit, ptr,
-                       local_size[1] * unit);
-                ptr += local_size[1] * unit;
-            }
+            // Tang
```
Comment (jeanbez) [inline src/api/pdc_region/pdc_region_transfer.c]: Can we drop this line and possibly the commented-out lines?

File: src/api/pdc_region/pdc_region_transfer.c [hunk 2]
```diff
@@ -1787,25 +1787,29 @@ release_region_buffer(char *buf, uint64_t *obj_dims, int local_ndim, uint64_t *l
...
-            for (i = 0; i < local_size[0]; ++i) {
-                for (j = 0; j < local_size[1]; ++j) {
-                    memcpy(buf + ((local_offset[0] + i) * obj_dims[1] * obj_dims[2] +
-                                  (local_offset[1] + j) * obj_dims[2] + local_offset[2]) *
-                                     unit,
-                           ptr, local_size[2] * unit);
-                    ptr += local_size[2] * unit;
-                }
-            }
+            // Tang
```
Comment (jeanbez) [inline src/api/pdc_region/pdc_region_transfer.c]: Can we drop this line and possibly the commented-out lines?

File: src/server/pdc_server_region/pdc_server_data.c [hunk 3]
```diff
@@ -4818,7 +4868,7 @@ PDC_Server_data_write_out(uint64_t obj_id, struct pdc_region_info *region_info,
                 // No need to update metadata
             }
             else {
-
+                // data_sieving_g = 1 by default, export PDC_DATA_SIEVING=0 to change
```
Comment (jeanbez) [inline src/server/pdc_server_region/pdc_server_data.c:4871]: We need to also change the documentation if this is the default now.

File: src/server/pdc_server_region/pdc_server_data.c [hunk 4]
```diff
@@ -5013,7 +5064,44 @@ PDC_Server_data_write_out(uint64_t obj_id, struct pdc_region_info *region_info,
     }
     if (is_contained == 0) {
         request_region->offset = lseek(region->fd, 0, SEEK_END);
-// printf("posix write for position %d with write size %u\n", 0, (unsigned)write_size);
+
+        // debug
+        /* fprintf(stderr, "==PDC_SERVER[%d]: %s posix write for position %d with write size %u\n", */
+        /*         pdc_server_rank_g, __func__, 0, (unsigned)write_size); */
+#ifdef ENABLE_ZFP
```
Comment (jeanbez) [inline src/server/pdc_server_region/pdc_server_data.c:5071]: This compression part is only relevant to the EQSIM code, or can it be used by any other test/code? Can we add something in the documentation for either case, including the option we need to enable/disable this?

File: src/server/pdc_server_region/pdc_server_region_cache.c [hunk 5]
```diff
@@ -6,10 +6,10 @@
 
 #ifdef PDC_SERVER_CACHE
 
-#ifdef PDC_SERVER_CACHE_MAX_SIZE
-#define MAX_CACHE_SIZE PDC_SERVER_CACHE_MAX_GB * 1024 * 1024 * 1024
+#ifdef PDC_SERVER_CACHE_MAX_GB
+#define MAX_CACHE_SIZE_GB PDC_SERVER_CACHE_MAX_GB
```
Comment (jeanbez) [inline src/server/pdc_server_region/pdc_server_region_cache.c:10]: We need to make sure the docs are updated correctly with the env name and unit people should use.

File: src/server/pdc_server_region/pdc_server_region_cache.c [hunk 6]
```diff
@@ -730,6 +739,11 @@ PDC_region_cache_flush_by_pointer(uint64_t obj_id, pdc_obj_cache *obj_cache, int
 #ifdef PDC_TIMING
     double start_time = MPI_Wtime();
 #endif
+    env_char = getenv("PDC_SERVER_CACHE_NO_FLUSH");
```
Comment (jeanbez) [inline src/server/pdc_server_region/pdc_server_region_cache.c:736]: Was the doc updated for this env?

**Review by jeanbez — CHANGES_REQUESTED** [review]

**Review by jeanbez — APPROVED** [review]

**Review by zhangwei217245 — APPROVED** [review]

**Pull request merged**